modules/fundamental/src/sbom/service/label.rs

```diff
@@ -1,13 +1,40 @@
-use crate::{Error, sbom::service::SbomService};
+use crate::{
+    Error,
+    sbom::{model::raw_sql::SEARCH_LABELS_SQL, service::SbomService},
+};
 use sea_orm::{
-    ActiveModelTrait, ActiveValue::Set, ConnectionTrait, DatabaseBackend, EntityTrait,
-    IntoActiveModel, QueryTrait, TransactionTrait,
+    ActiveModelTrait, ActiveValue::Set, ConnectionTrait, DatabaseBackend, DbBackend, EntityTrait,
+    FromQueryResult, IntoActiveModel, PaginatorTrait, QueryTrait, Statement, TransactionTrait,
 };
 use sea_query::Expr;
 use trustify_common::id::{Id, TrySelectForId};
 use trustify_entity::{labels::Labels, sbom};
 
 impl SbomService {
+    // Fetch all unique key/value labels of all SBOMs
+    // If limit=0 then all data will be fetched
+    pub async fn fetch_labels<C: ConnectionTrait>(
+        &self,
+        filter_text: &str,
+        limit: u64,
+        connection: &C,
+    ) -> Result<Vec<serde_json::Value>, Error> {
+        let statement = Statement::from_sql_and_values(
+            DbBackend::Postgres,
+            SEARCH_LABELS_SQL,
+            [format!("%{}%", filter_text).into()],
```

Comment:
That looks like a candidate for an SQL inject attack. Please use SQL parameters and maybe CONCAT.